lib/checks/isPolynomialToConstantPower.js

```diff
@@ -0,0 +1,19 @@
+const Node = require('../node');
+
+function isPolynomialToConstantPower(node) {
+  // For the special case when the node is in the form (x + 2)^2
```

Comment:
I think just to be super clear

"For the special case when the node is a polynomial to a constant power, e.g. (x+2)^2"

cause exactly (x+2)^2 is technically one special case, but you mean to catch more than that :p